### PR TITLE
Switch to new image_compare.py for testing

### PR DESCRIPTION
**Pull request by rcolyer**
Following up on #4462 and broader concerns, I wrote a new image comparison method with Python.  Upon completing it and running it, I found a fairly large number of tests that should have failed but the previous comparison method was not detecting the issue, as we had been using a number of pixels over a threshold of difference with ImageMagick.  Because of issues with small edge effects, the threshold was set very large, but this left the opportunity for some fairly significant differences to slip by.

This new test does not compare pixels, but checks all 3x3 pixel blocks (with overlap) for like-signed difference between actual and expected, and fails the test if any 3x3 block has this property.  None of the failing tests in my run were improperly failing, and all of the new fails were tests that should have been updated with prior changes.

Fortunately, no hidden errors in OpenSCAD were detected in this update to the testing, but we simply had some substantial alterations in output (on purpose) that we did not have adequate test awareness of (which undermines the utility of testing).  Hopefully this will give us more rigor in the future.

Of importance is that this update introduces generation of a venv to the cmake run, to setup numpy and PIL for the image_compare.py file.  Provided it actually ends up eventually passing correctly on all CI platforms, I think this is likely a net benefit to the codebase.  This opens up future option of a rich set of Python libraries for more elaborate statistical tests, or flexible customized image testing if a particular future test happens to not work well with the standard image comparison.

This update also removes the ImageMagick dependency requirement for running tests.  However the previous behavior of ImageMagick or diffpng remains available as an option with:  cmake -DUSE_IMAGE_COMPARE_PY=OFF

### CODE REVIEW

**Comment by rcolyer** [comment]
Note that the remaining failing test, that happens on only the macos automated testing, is catching a real defect in the produced test result for the issue3158 test.  (Which rather than a fail for the PR is the point of it.)

File: tests/image_compare.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+from PIL import Image
+import numpy as np
+import sys
+
+
+def Compare3x3(img1, img2):
```
Comment (kintel) [inline tests/image_compare.py:8]: It wouldn't hurt with a short explanation what this actually does, as this code is pretty hard to read.

**Review by kintel — APPROVED** [review]

File: tests/image_compare.py (hunk 1 above)
Comment (rcolyer) [reply tests/image_compare.py:8]: Right.  Done.

Comment (kintel) [reply tests/image_compare.py:8]: Awesome!

**Pull request merged**